### PR TITLE
docs(@core/home/hero): add documentation

### PR DESCRIPTION
**Pull request by marobakr**
1. add documentation
if applied, it will make code easier to maintain and readable

### CODE REVIEW

File: src/app/core/components/home/hero/hero.component.ts [hunk 1]
```diff
@@ -2,30 +2,51 @@ import { BreakpointObserver, Breakpoints } from '@angular/cdk/layout';
...
 export class HeroComponent {
+  /**
+   * Flag indicating whether the viewport is in handset portrait mode.
+   * @type {boolean}
+   */
   isHandsetPortrait: boolean = false;
 
   constructor(private _breakpointObserver: BreakpointObserver) {}
 
+  /**
```
Comment (vcpc) [inline src/app/core/components/home/hero/hero.component.ts:25]: This is a built-in method, and it's well-known, only documentate code made by you.

File: src/app/core/components/home/hero/hero.component.ts [hunk 2]
```diff
@@ -2,30 +2,51 @@ import { BreakpointObserver, Breakpoints } from '@angular/cdk/layout';
...
+   * Initialize the component and start observing the handset portrait breakpoint.
+   * @returns {void}
+   */
   ngOnInit(): void {
     this.HandsetPortraitBreakpoint();
   }
 
+  /**
+   * Observe the handset portrait breakpoint and update the `isHandsetPortrait` flag accordingly when the user change display at screen.
```
Comment (vcpc) [inline src/app/core/components/home/hero/hero.component.ts:34]: good, but next time type `HandsetPortrait` (in a codespan) breakpoint, so when anyone reads it realizes that this is a part from the code not a some normal text

File: src/app/core/components/home/hero/hero.component.ts [hunk 3]
```diff
@@ -2,30 +2,51 @@ import { BreakpointObserver, Breakpoints } from '@angular/cdk/layout';
...
   standalone: true,
   imports: [CommonModule, RouterLink],
   templateUrl: './hero.component.html',
-  styleUrl: './hero.component.scss',
+  styleUrls: ['./hero.component.scss'],
 })
 export class HeroComponent {
+  /**
+   * Flag indicating whether the viewport is in handset portrait mode.
+   * @type {boolean}
```
Comment (vcpc) [inline src/app/core/components/home/hero/hero.component.ts:19]: types are defined by TypeScript, no need to documentate it.
same thind at lines: 27, 35, 48

**Comment by marobakr** [comment]
1. add documentation
1.1 add documentation for built-in method "constructor"
1.2 wrapping HandsetPortrait by dubble codes
1.3 remove useless type definition 

if applied, it will make code easier to maintain and readable

**Comment by marobakr** [comment]
1. add documentation
1.2 wrapping any flags in text with (codespan)
1.3 remove useless type definition

if applied, it will make code easier to maintain and readable

**Pull request merged**